### ISSUE TITLE
NMT

### ISSUE THREAD

**Issue by dikshant2210** [jellAIfish/jellyfish#16]
[Neural Machine Translation by jointly learning to align and translate](https://arxiv.org/abs/1409.0473)

**Comment by dikshant2210** [comment]
Unlike the traditional statistical machine translation, the neural machine translation aims at building a single neural network that can be jointly tuned to maximize the translation performance. In this paper, we conjecture that the use of a fixed-length vector is a bottleneck in improving the performance of this basic encoder–decoder architecture, and propose to extend this by allowing a model to automatically (soft-)search for parts of a source sentence that are relevant to predicting a target word, without having to form these parts as a hard segment explicitly.

An encoder neural network reads and encodes a source sentence into a fixed-length vector. A decoder then outputs a translation from the encoded vector. A potential issue with this encoder–decoder approach is that a neural network needs to be able to compress all the necessary information of a source sentence into a fixed-length vector. 
The performance of a basic encoder–decoder deteriorates rapidly as the length of an input sentence increases. Each time the proposed model generates a word in a translation, it (soft-)searches for a set of positions in a source sentence where the most relevant information is concentrated. 
The model then predicts a target word based on the context vectors associated with these source positions and all the previous generated target words. 
The most important distinguishing feature of this approach from the basic encoder–decoder is that
it does not attempt to encode a whole input sentence into a single fixed-length vector.

It should be noted that unlike the existing encoder–decoder approach, here the probability is conditioned on a distinct context vector ci for each target word yi.

The context vector ci depends on a sequence of annotations to which an encoder maps the input sentence. Each(h1, · · · ,hTx) to which an encoder maps the input sentence. Each annotation hi (h1, · · · ,hTx) to which an encoder maps the input sentence. Each annotation hi contains information about the whole input sequence with a strong focus on the parts surrounding the i-th word of the input sequence.

The alignment model directly com-putes a soft alignment, which allows the gradient of the cost function to be backpropagated through. This gradient can be used to train the alignment model as well as the whole translation model jointly.
